### ISSUE TITLE
Am I seeing a memory leak with UCP?

### ISSUE THREAD

**Issue by rrgargeya** [openucx/ucx#9114]
I see unfreed memory messages, given in the attached file, when my application ends.

I closed out all the UCP objects properly before the application exits.

Are these messages okay or am I missing the proper closing of some UCP objects.

I am concerned that there could be a memory leak that over time creates garbage that may not collected in
time.
[ucx_ticket3_memmgmt_attach1.log](https://github.com/openucx/ucx/files/11629865/ucx_ticket3_memmgmt_attach1.log)

**Comment by brminich** [comment]
did you call `ucp_mem_unmap` for the corresponding `ucp_mem_map` calls?

**Comment by rrgargeya** [comment]
Yes, I called ucp_mem_unmap() corresponding to the ucp_mem_map().

I see identical logs even in programs that did not have any memory mapping.

I made sure that I closed all endpoints, and destroyed the workers.

These log messages happen right after I clean up the application context (ucp_cleanup()), indicating there were other objects that were perhaps internally created that were causing these messages.

Is there a way to close them or clean them up or are these messages inevitable?
